### PR TITLE
[Layout foundations] Update Columns component docs and guidance

### LINKED ISSUE #6906: [Layout foundations] Update `Columns` component docs and guidance
Update `Columns` docs

- Description
- Examples: descriptions and visuals
- Add related components section

| Update | |
|---|---|
| Description | Displays content horizontally in one or more columns with equal spacing between. |
| Default example description | By default, there are 6 columns with 16px spacing between them. |
| Spacing example description | Use the spacing prop to set the amount of space between columns. |
| Column width example description | Individual column width can be adjusted. |
| Number of columns example description | Use the columns prop to set the number and width of columns across breakpoints. |
| Related components | For more control over padding and widths, use the Box component|
| | To lay out a set of smaller components horizontally, use the Inline component |

**Before and after**

Image

Link to [Figma](https://www.figma.com/file/5oEi1wm73KhGLN0VS02oYm/Layout-components?node-id=2764%3A32219)

### PR DESCRIPTION
**Pull request by chazdean**
### WHY are these changes introduced?

Fixes #6906  

### WHAT is this pull request doing?

Updates the `Columns` component docs and guidance

 <details>
      <summary>Columns style guide</summary>
     Columns styleguide
    </details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by sarahill — APPROVED** [review]
One thing I was wondering with the `AlphaStack` examples and now with these, is if we always want to highlight spacing between or just for certain examples? 

Seeing it, I'm trying to figure out why only certain ones are highlighted but it doesn't feel clear. Not a blocker just something for us to figure out. @nayeob do you have thoughts?

File: polaris.shopify.com/content/components/columns.md [hunk 1]
```diff
@@ -10,8 +10,23 @@ status:
...
+      Individual column width can be adjusted.
+  - fileName: columns-with-set-number.tsx
+    title: Number of columns
+    description: >-
+      Use the columns prop to set the number and width of columns across breakpoints.
 ---
+
+## Related components
+
+- For more control over padding and widths, use the [Box](https://polaris.shopify.com/components/box) component
```
Comment (sarahill) [inline polaris.shopify.com/content/components/columns.md]:
```suggestion
- For more control over padding and widths, [use the Box](https://polaris.shopify.com/components/box) component
```

File: polaris.shopify.com/content/components/columns.md [hunk 2]
```diff
@@ -10,8 +10,23 @@ status:
...
+- To lay out a set of smaller components horizontally, use the [Inline](https://polaris.shopify.com/components/inline) component
```
Comment (sarahill) [inline polaris.shopify.com/content/components/columns.md]:
```suggestion
- To lay out a set of smaller components horizontally, [use the Inline](https://polaris.shopify.com/components/inline) component
```

File: polaris.shopify.com/pages/examples/columns-default.tsx [hunk 3]
```diff
@@ -1,21 +1,45 @@
...
+  return (
+    <div
+      style={{
+        display: 'inherit',
+        background: '#7B47F1',
+        height: height ?? undefined,
+        width: width ?? undefined,
+      }}
+    >
+      <Inline align={'center'} alignY={'center'}>
```
Comment (laurkim) [inline polaris.shopify.com/pages/examples/columns-default.tsx]:
```suggestion
      <Inline align="center" alignY="center">
```

Small nit

File: polaris.shopify.com/pages/examples/columns-with-fixed-widths.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  return (
+    <div
+      style={{
+        display: 'inherit',
+        background: '#7B47F1',
+        height: height ?? undefined,
+        width: width ?? undefined,
+      }}
+    >
+      <Inline align={'center'} alignY={'center'}>
```
Comment (laurkim) [inline polaris.shopify.com/pages/examples/columns-with-fixed-widths.tsx]:
```suggestion
      <Inline align="center" alignY="center">
```

Small nit

File: polaris.shopify.com/pages/examples/columns-with-set-number.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  return (
+    <div
+      style={{
+        display: 'inherit',
+        background: '#7B47F1',
+        height: height ?? undefined,
+        width: width ?? undefined,
+      }}
+    >
+      <Inline align={'center'} alignY={'center'}>
```
Comment (laurkim) [inline polaris.shopify.com/pages/examples/columns-with-set-number.tsx]:
```suggestion
      <Inline align="center" alignY="center">
```

Small nit

File: polaris.shopify.com/pages/examples/columns-with-varying-spacing.tsx [hunk 6]
```diff
@@ -1,21 +1,62 @@
...
+  return (
+    <div
+      style={{
+        display: 'inherit',
+        background: '#7B47F1',
+        height: height ?? undefined,
+        width: width ?? undefined,
+      }}
+    >
+      <Inline align={'center'} alignY={'center'}>
```
Comment (laurkim) [inline polaris.shopify.com/pages/examples/columns-with-varying-spacing.tsx]:
```suggestion
      <Inline align="center" alignY="center">
```

Small nit

**Review by laurkim — APPROVED** [review]
This looks great! 🚀

**Pull request merged**